engine/src/rgbmatrix.cpp

```diff
@@ -166,8 +173,15 @@ bool RGBMatrix::copyFrom(const Function* function)
         setAlgorithm(mtx->algorithm()->clone());
     else
         setAlgorithm(NULL);
-    setStartColor(mtx->startColor());
-    setEndColor(mtx->endColor());
+
+    QVectorIterator<QColor> it(mtx->getColors());
```

Comment:
foreach (QColor col, mtx->getColors())
        m_rgbColors.append(col);

But it won't apply the colors to the algorithm.
Why isn't there a m_algorithm->setColors(m_rgbColors) in the setAlgorithm call?

Reply:
In line 304, setStartColor() is replaced by a generic setColor(), which here is called a few lines later after your commented line (line 181) while iterating through the colors. That then sets the colors to m_algorithm->setColors in line 313.